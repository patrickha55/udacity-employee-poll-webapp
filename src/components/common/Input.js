/**
 * @param {{
 *  id: string,
 *  label: string,
 *  name: string,
 *  additionalClassNames: string,
 *  handleChange: function,
 *  handleBlur: function,
 *  value: string,
 *  placeHolder: string,
 *  type: string,
 * }} props
 * 
 * @returns {JSX.Element}
 */
const Input = ({
  id,
  label,
  name,
  additionalClassNames = '',
  handleChange,
  handleBlur,
  value,
  isRequired = false,
  placeHolder = '',
  type,
}) => {
  return (
    <>
      <label htmlFor={id} className="form-label">{label}</label>
      {
        isRequired === true ? (
          <input
            type={type}
            name={name}
            className={`form-control ${additionalClassNames}`}
            id={id}
            onChange={handleChange}
            onBlur={handleBlur}
            value={value}
            placeholder={placeHolder}
            autoComplete='on'
            data-testid={`${name}RequiredInput`}
            required
          />
        ) : (
          <input
            type={type}
            name={name}
            className={`form-control ${additionalClassNames}`}
            id={id}
            onChange={handleChange}
            onBlur={handleBlur}
            placeholder={placeHolder}
            value={value}
            data-testid={`${name}Input`}
            autoComplete='on'
          />
        )
      }
    </>
  );
};

export default Input;