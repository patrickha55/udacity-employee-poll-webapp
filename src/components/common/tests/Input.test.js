import { render, screen } from '@testing-library/react';
import Input from '../Input';
import renderer from 'react-test-renderer';

describe('Test block for Input component', () => {
  it('should renders snapshot', () => {
    const component = renderer.create(
      <Input
        name='testInput'
        type='text'
        label='Test Input'
        id='test'
        handleChange={() => { }}
        handleBlur={() => { }}
        value='test'
        isRequired={true}
        placeHolder='test'
      />
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should render Input component', () => {
    render(
      <Input
        name='testInput'
        type='text'
        label='Test Input'
        id='test'
        handleChange={() => { }}
        handleBlur={() => { }}
        value='test'
        isRequired={true}
        placeHolder='test'
      />
    );

    const inputLabel = screen.getByText(/test input/i);
    const input = screen.getByPlaceholderText('test');
    const inputByRole = screen.getByRole('textbox');

    expect(inputLabel).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(inputByRole).toBeInTheDocument();
    expect(input.value).toBe('test');
  });
});
