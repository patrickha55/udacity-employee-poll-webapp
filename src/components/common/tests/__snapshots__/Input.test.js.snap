// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test block for Input component should renders snapshot 1`] = `
Array [
  <label
    className="form-label"
    htmlFor="test"
  >
    Test Input
  </label>,
  <input
    autoComplete="on"
    className="form-control "
    data-testid="testInputRequiredInput"
    id="test"
    name="testInput"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="test"
    required={true}
    type="text"
    value="test"
  />,
]
`;
